### PR TITLE
Data model: add extra information for run 2 evsel

### PR DESCRIPTION
**Pull request by ddobrigk**
Goes hand-in-hand with: https://github.com/alisw/AliPhysics/pull/23817

* adds extra event selection information to the run 2 converted data by adding:
   * TPCout flag in track flags 
   * total number of tpc clusters as part of run2bcinfos
   * total number of SSD+SDD clusters as part of run2bcinfos

Run2BCInfos now becomes versioned, Run 2 converter in AliPhysics will immediately generate version `001`. An O2Physics converter will be added tomorrow. 

For reference, data size increase is negligible (`Run2BCInfos` currently occupies, with this change, 0.2% of the data size even in pp, where it's smallest).

### CODE REVIEW

**Review by victor-gonzalez — COMMENTED** [review]
@ddobrigk Perfect! Thanks!

File: Framework/Core/include/Framework/AnalysisDataModel.h [hunk 1]
```diff
@@ -1721,12 +1723,23 @@ DECLARE_SOA_COLUMN(Mass, mass, float);                                  //! mass
...
+DECLARE_SOA_TABLE_VERSIONED(Run2BCInfos_001, "AOD", "RUN2BCINFO", 1,
+                            run2::EventCuts, //! Legacy information for Run 2 event selection
+                            run2::TriggerMaskNext50, run2::L0TriggerInputMask,
+                            run2::SPDClustersL0, run2::SPDClustersL1,
+                            run2::SPDFiredChipsL0, run2::SPDFiredChipsL1,
+                            run2::SPDFiredFastOrL0, run2::SPDFiredFastOrL1,
+                            run2::V0TriggerChargeA, run2::V0TriggerChargeC,
+                            run2::NTPCClusters, run2::NSDDSSDClusters);
+
+using Run2BCInfos = Run2BCInfos_000;
```
Comment (victor-gonzalez) [inline Framework/Core/include/Framework/AnalysisDataModel.h:1742]: @ddobrigk Shouldn't be the `001` in place if there are going to be a converter?

Comment (ddobrigk) [reply Framework/Core/include/Framework/AnalysisDataModel.h:1742]: @victor-gonzalez no, step 1 is to keep using vwrsion 000, then step 2 is to add the converter, and then step 3 is to switch to version 001. This is as intended :-)

**Comment by pzhristov** [comment]
@ddobrigk Should I merge it?

**Comment by ddobrigk** [comment]
@pzhristov yes, please! Thanks! I have the converter ready for tomorrow, too.

**Pull request merged**